### DISCUSSION TITLE
Firebase Analytics Alternative

### DISCUSSION THREAD

**Discussion by vHanda** [supabase/supabase#793] [Feature Requests]
Hello. Right now Firebase Analytics, even the client side libraries, are completely proprietary. Having proper analytics really lets you understand how your users are currently using your product.

I'd love for Supabase to tackle this as well. Maybe just start with a simple dump to postgres, which users can explore via Grafana / other-solutions. And then overtime go towards exposing the default dashboard. I understand that this takes you down a very deep rabbit hole, but just providing the basics would be wonderful.

There are big lack of open source mobile analytics solutions.

**Comment by cdedreuille** [comment]
Mixpanel serves us quite well when it comes to event tracking. It's way better than Firebase Analytics. If you add that to the generic Google Analytics you have a pretty similar setup.

**Comment by kiwicopple** [comment]
Great suggestion. You're one of the first to request this - most commonly devs want the Storage and Functions which we are working on now. I'm curious to see if this gets interest/upvotes. 

If anyone else has some alternatives/suggestions or Prior Art to share on this topic, that would also be welcomed.

**Comment by ilhamwahabi** [comment]
For client side analytics, I think I would recommend Supabase using https://plausible.io/ behind the screen. You can retrieve the data using API. It's also open source and have small sizes (0.4kb) so It really have same spirit

**Reply by vHanda** [reply]
I spent some time looking into this over the weekend. In the end, I wouldn't recommend Plausible (backend) as -

* AGPL licensed - This could be an issue
* Their database schema is built around website analytics, and pigeonholing all of firebase's analytics data into it wouldn't be optimal. For example - associating userIDs with analytics isn't something they do, by design.
* Their UIs are also not built for exploring the data and are very web-centric instead of app-centric.
* Codebase is based on Elixir (harder for find programmers) 

Regarding the client side - Plausible is based around the concept of not collecting any user data and not requiring the GDPR or cookies consent dialog. Whereas in an application, I am collecting user data, and I want to also attach analytics to that.

Edit: I don't mean to paint Plausible in a bad light. It's a wonderful project, and I use it for my websites. It just doesn't seem like it would add anything when building an alternative to Firebase analytics.

**Comment by Shabinder** [comment]
+1 for alternative to firebase analytics & Crashlytics

**Comment by srvmux** [comment]
wouldn't something like this work? [countly pricing](https://count.ly/pricing) it can be self host and it has crash reporting, mobile and web SDK's on the free version but the enterprise version has a lot more stuff, but I personally never tried it.

**Reply by gomflo** [reply]
umami is another option

https://github.com/mikecao/umami

**Reply by srvmux** [reply]
> umami is another option
> 
> https://github.com/mikecao/umami

seems just like plausible imo

**Reply by vHanda** [reply]
The free version is missing basic stuff like revenue and retention tracking.

**Reply by srvmux** [reply]
> The free version is missing basic stuff like revenue and retention tracking.

true and there's also [matomo](https://matomo.org/integrate/)
but it looks like all of their integrations are community maintained and some of them don't look too promising

**Comment by eykrehbein** [comment]
Hey folks, I'm currently building an user analytics service that directly integrates with Supabase using Postgres Triggers. 
It's not open-source but it's going to have a generous free plan. If you're interested, you can head over to [userstats.io](https://userstats.io) to get early-access. I'd also love to hear your thoughts on it, including what stats you'd like to see in a dashboard!

**Reply by vHanda** [reply]
Honestly, it not being open source, immediately rules it out, as integrating with Postgres isn't the point. And then I'd go over with something like Mix Panel.

Also, from your website, it's really not clear what UserStats.io is providing apart "using Postgres Triggers", and a few basic statistics.

**Reply by eykrehbein** [reply]
Thanks for your feedback! What are the main reasons you want it to be open-source? Is it lack of trust in the integration/service or more the whole concept of open source so you can contribute etc.?

**Reply by vHanda** [reply]
avoiding vendor lock-in and ease of customization.
